### ISSUE TITLE
Calendar widget for iOS overflown by 5 pixels

### ISSUE THREAD

**Issue by DhruvamSharma** [flutter/flutter#32991]
## Steps to Reproduce

1. Use showDatePicker function as shown in the docs: https://api.flutter.dev/flutter/material/showDatePicker.html
2. Run the app on iOS XR- 12.1
3. The picker's bottom overflows by 5.0 Pixels.

![Simulator Screen Shot - iPhone XR - 2019-05-19 at 15 44 16](https://user-images.githubusercontent.com/18664195/57980866-52ff1f80-7a4e-11e9-8090-cd8e37210f8b.png)

## Logs

```
[        ] Updating files
[ +316 ms] DevFS: Sync finished
[        ] Syncing files to device iPhone XR... (completed in 4,123ms, longer than expected)
[   +1 ms] Synced 1.3MB.
[   +1 ms] Sending to VM service: _flutter.listViews({})
[   +2 ms] Result: {type: FlutterViewList, views: [{type: FlutterView, id: _flutterView/0x7f936b0bea18, isolate: {type: @Isolate, fixedId: true, id: isolates/345817298, name:
main.dart$main-345817298, number: 345817298}}]}
[        ] <- accept
[        ] Connected to _flutterView/0x7f936b0bea18.
[   +3 ms] 🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
[   +1 ms] An Observatory debugger and profiler on iPhone XR is available at: http://127.0.0.1:55190/w3out7g3hxc=/
[        ] For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
[+22089 ms] [DEVICE LOG] 2019-05-19 16:00:18.923806+0530  localhost Runner[32361]: (Flutter) flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY
╞═════════════════════════════════════════════════════════
[  +13 ms] flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
[        ] [DEVICE LOG] 2019-05-19 16:00:18.931207+0530  localhost Runner[32361]: (Flutter) flutter: The following message was thrown during layout:
[        ] [DEVICE LOG] 2019-05-19 16:00:18.931354+0530  localhost Runner[32361]: (Flutter) flutter: A RenderFlex overflowed by 5.0 pixels on the bottom.
[        ] flutter: The following message was thrown during layout:
[        ] flutter: A RenderFlex overflowed by 5.0 pixels on the bottom.
[  +10 ms] [DEVICE LOG] 2019-05-19 16:00:18.947284+0530  localhost Runner[32361]: (Flutter) flutter:
[        ] [DEVICE LOG] 2019-05-19 16:00:18.947380+0530  localhost Runner[32361]: (Flutter) flutter: The overflowing RenderFlex has an orientation of Axis.vertical.
[        ] [DEVICE LOG] 2019-05-19 16:00:18.947656+0530  localhost Runner[32361]: (Flutter) flutter: The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow
and
[        ] [DEVICE LOG] 2019-05-19 16:00:18.947989+0530  localhost Runner[32361]: (Flutter) flutter: black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
[        ] [DEVICE LOG] 2019-05-19 16:00:18.948379+0530  localhost Runner[32361]: (Flutter) flutter: Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
[        ] [DEVICE LOG] 2019-05-19 16:00:18.948814+0530  localhost Runner[32361]: (Flutter) flutter: RenderFlex to fit within the available space instead of being sized to their natural size.
[        ] flutter:
[        ] flutter: The overflowing RenderFlex has an orientation of Axis.vertical.
[        ] flutter: The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
[   +3 ms] flutter: black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
[        ] flutter: Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
[        ] flutter: RenderFlex to fit within the available space instead of being sized to their natural size.
[        ] [DEVICE LOG] 2019-05-19 16:00:18.949306+0530  localhost Runner[32361]: (Flutter) flutter: This is considered an error condition because it indicates that there is content that cannot be
[        ] [DEVICE LOG] 2019-05-19 16:00:18.949772+0530  localhost Runner[32361]: (Flutter) flutter: seen. If the content is legitimately bigger than the available space, consider clipping it with
a
[        ] [DEVICE LOG] 2019-05-19 16:00:18.949954+0530  localhost Runner[32361]: (Flutter) flutter: ClipRect widget before putting it in the flex, or using a scrollable container rather than a
Flex,
[        ] [DEVICE LOG] 2019-05-19 16:00:18.950226+0530  localhost Runner[32361]: (Flutter) flutter: like a ListView.
[        ] [DEVICE LOG] 2019-05-19 16:00:18.950436+0530  localhost Runner[32361]: (Flutter) flutter: The specific RenderFlex in question is:
[        ] [DEVICE LOG] 2019-05-19 16:00:18.950562+0530  localhost Runner[32361]: (Flutter) flutter:   RenderFlex#8313f OVERFLOWING
[        ] [DEVICE LOG] 2019-05-19 16:00:18.951246+0530  localhost Runner[32361]: (Flutter) flutter:   creator: Column ← Padding ← DecoratedBox ← ConstrainedBox ← Container ← _DatePickerHeader ←
Column
[   +2 ms] [DEVICE LOG] 2019-05-19 16:00:18.951643+0530  localhost Runner[32361]: (Flutter) flutter:   ← SizedBox ← LayoutBuilder ← OrientationBuilder ← DefaultTextStyle ← AnimatedDefaultTextStyle
← ⋯
[        ] [DEVICE LOG] 2019-05-19 16:00:18.951815+0530  localhost Runner[32361]: (Flutter) flutter:   parentData: offset=Offset(16.0, 0.0) (can use size)
[        ] [DEVICE LOG] 2019-05-19 16:00:18.951959+0530  localhost Runner[32361]: (Flutter) flutter:   constraints: BoxConstraints(w=298.0, h=100.0)
[        ] [DEVICE LOG] 2019-05-19 16:00:18.952591+0530  localhost Runner[32361]: (Flutter) flutter:   size: Size(298.0, 100.0)
[        ] [DEVICE LOG] 2019-05-19 16:00:18.952774+0530  localhost Runner[32361]: (Flutter) flutter:   direction: vertical
[        ] [DEVICE LOG] 2019-05-19 16:00:18.952935+0530  localhost Runner[32361]: (Flutter) flutter:   mainAxisAlignment: center
[   +5 ms] [DEVICE LOG] 2019-05-19 16:00:18.953185+0530  localhost Runner[32361]: (Flutter) flutter:   mainAxisSize: max
[        ] flutter: This is considered an error condition because it indicates that there is content that cannot be
[        ] flutter: seen. If the content is legitimately bigger than the available space, consider clipping it with a
[        ] flutter: ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
[        ] flutter: like a ListView.
[        ] flutter: The specific RenderFlex in question is:
[        ] flutter:   RenderFlex#8313f OVERFLOWING
[        ] flutter:   creator: Column ← Padding ← DecoratedBox ← ConstrainedBox ← Container ← _DatePickerHeader ← Column
[        ] flutter:   ← SizedBox ← LayoutBuilder ← OrientationBuilder ← DefaultTextStyle ← AnimatedDefaultTextStyle ← ⋯
[   +4 ms] flutter:   parentData: offset=Offset(16.0, 0.0) (can use size)
[        ] flutter:   constraints: BoxConstraints(w=298.0, h=100.0)
[        ] flutter:   size: Size(298.0, 100.0)
[        ] flutter:   direction: vertical
[        ] flutter:   mainAxisAlignment: center
[        ] flutter:   mainAxisSize: max
[        ] [DEVICE LOG] 2019-05-19 16:00:18.953328+0530  localhost Runner[32361]: (Flutter) flutter:   crossAxisAlignment: start
[        ] [DEVICE LOG] 2019-05-19 16:00:18.953436+0530  localhost Runner[32361]: (Flutter) flutter:   textDirection: ltr
[        ] [DEVICE LOG] 2019-05-19 16:00:18.953750+0530  localhost Runner[32361]: (Flutter) flutter:   verticalDirection: down
[        ] [DEVICE LOG] 2019-05-19 16:00:18.954389+0530  localhost Runner[32361]: (Flutter) flutter:
◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
[   +7 ms] [DEVICE LOG] 2019-05-19 16:00:18.955226+0530  localhost Runner[32361]: (Flutter) flutter:
════════════════════════════════════════════════════════════════════════════════════════════════════
[   +3 ms] flutter:   crossAxisAlignment: start
[        ] flutter:   textDirection: ltr
[        ] flutter:   verticalDirection: down
[        ] flutter: ◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
[        ] flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════


```

Flutter Analyze:
```
Analyzing amspaces...                                                   
No issues found! (ran in 5.6s)

```

Flutter Doctor -v

```
[✓] Flutter (Channel beta, v1.5.4-hotfix.2, on Mac OS X 10.14.3 18D109, locale en-IN)
    • Flutter version 1.5.4-hotfix.2 at /Users/dhruvamsharma/flutter
    • Framework revision 7a4c33425d (3 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

 
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/dhruvamsharma/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.

 
[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ libimobiledevice and ideviceinstaller are not installed. To install with Brew, run:
        brew update
        brew install --HEAD usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.4
    • CocoaPods version 1.6.1

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] Connected device (1 available)
    • iPhone XR • BDB1717C-4372-4130-B213-22D96094C827 • ios • iOS 12.1 (simulator)

! Doctor found issues in 2 categories.

```

**Comment by AiHMin** [comment]
This problem has been fixed by PR #31514. You might switch into dev channel and use the latest dev version `1.6.0`. It should work.

**Comment by DhruvamSharma** [comment]
Thank you @AiHMin  :)
